src/main/java/org/apache/commons/lang3/ClassUtils.java

```diff
@@ -0,0 +1,1727 @@
+     * for these cases.
+     * </p>
+     *
+     * <p>
+     * {@code Null} may be assigned to any reference type. This method will return {@code true} if {@code null} is passed in
+     * and the toClass is non-primitive.
+     * </p>
+     *
+     * <p>
+     * Specifically, this method tests whether the type represented by the specified {@code Class} parameter can be
+     * converted to the type represented by this {@code Class} object via an identity conversion widening primitive or
+     * widening reference conversion. See <em><a href="http://docs.oracle.com/javase/specs/">The Java Language
+     * Specification</a></em>, sections 5.1.1, 5.1.2 and 5.1.4 for details.
+     * </p>
+     *
+     * <p>
+     * <strong>Since Lang 3.0,</strong> this method will default behavior for calculating assignability between primitive
+     * and wrapper types <em>corresponding to the running Java version</em>; i.e. autoboxing will be the default behavior in
+     * VMs running Java versions &gt; 1.5.
+     * </p>
+     *
+     * @param cls the Class to check, may be null
+     * @param toClass the Class to try to assign into, returns false if null
+     * @return {@code true} if assignment possible
+     */
+    public static boolean isAssignable(final Class<?> cls, final Class<?> toClass) {
+        return isAssignable(cls, toClass, true);
+    }
+
+    /**
+     * <p>
+     * Checks if one {@code Class} can be assigned to a variable of another {@code Class}.
+     * </p>
+     *
+     * <p>
+     * Unlike the {@link Class#isAssignableFrom(java.lang.Class)} method, this method takes into account widenings of
+     * primitive classes and {@code null}s.
+     * </p>
+     *
+     * <p>
+     * Primitive widenings allow an int to be assigned to a long, float or double. This method returns the correct result
+     * for these cases.
+     * </p>
+     *
+     * <p>
+     * {@code Null} may be assigned to any reference type. This method will return {@code true} if {@code null} is passed in
+     * and the toClass is non-primitive.
+     * </p>
+     *
+     * <p>
+     * Specifically, this method tests whether the type represented by the specified {@code Class} parameter can be
+     * converted to the type represented by this {@code Class} object via an identity conversion widening primitive or
+     * widening reference conversion. See <em><a href="http://docs.oracle.com/javase/specs/">The Java Language
+     * Specification</a></em>, sections 5.1.1, 5.1.2 and 5.1.4 for details.
+     * </p>
+     *
+     * @param cls the Class to check, may be null
+     * @param toClass the Class to try to assign into, returns false if null
+     * @param autoboxing whether to use implicit autoboxing/unboxing between primitives and wrappers
+     * @return {@code true} if assignment possible
+     */
+    public static boolean isAssignable(Class<?> cls, final Class<?> toClass, final boolean autoboxing) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1349-1377 into a separate method.